File: src/Entity/Local.php
<?php

namespace App\Entity;

use App\Repository\LocalRepository;
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Serializer\Annotation\Groups;
use Symfony\Component\Validator\Constraints as Assert;
#[ORM\Entity(repositoryClass: LocalRepository::class)]
class Local
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[Groups("locals")]
    #[ORM\Column]
    private ?int $id = null;

    #[ORM\Column(length: 255)]
    #[Assert\NotBlank(message:"Le nom  du block est requis")]
    #[Groups("locals")]
    private ?string $nomBlock = null;

    #[ORM\Column(length: 255)]
    #[Assert\NotBlank(message:"Le nom du la patient est requis")]
    #[Groups("locals")]
    #[Assert\Length(min: 2,minMessage: "veuillez avoir au minimum 2 caractere" )]
    private ?string $nomPatient = null;

    #[ORM\Column(length: 255)]
    #[Assert\NotBlank(message:"Le nom  du la medecin est requis")]
    #[Groups("locals")]
    private ?string $nomMedecin = null;

    #[ORM\Column(length: 255)]
    #[Assert\NotBlank(message:"Le localisation est requis")]
    #[Groups("locals")]
    private ?string $localisation = null;

    #[ORM\OneToMany(mappedBy: 'locaux', targetEntity: Consultation::class)]
    private Collection $consultations;

    public function __construct()
    {
        $this->consultations = new ArrayCollection();
    }

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getNomBlock(): ?string
    {
        return $this->nomBlock;
    }

    public function setNomBlock(string $nomBlock): self
    {
        $this->nomBlock = $nomBlock;

        return $this;
    }

    public function getNomPatient(): ?string
    {
        return $this->nomPatient;
    }

    public function setNomPatient(string $nomPatient): self
    {
        $this->nomPatient = $nomPatient;

        return $this;
    }

    public function getNomMedecin(): ?string
    {
        return $this->nomMedecin;
    }

    public function setNomMedecin(string $nomMedecin): self
    {
        $this->nomMedecin = $nomMedecin;

        return $this;
    }

    public function getLocalisation(): ?string
    {
        return $this->localisation;
    }

    public function setLocalisation(string $localisation): self
    {
        $this->localisation = $localisation;

        return $this;
    }

    /**
     * @return Collection<int, Consultation>
     */
    public function getConsultations(): Collection
    {
        return $this->consultations;
    }

    public function addConsultation(Consultation $consultation): self
    {
        if (!$this->consultations->contains($consultation)) {
            $this->consultations->add($consultation);
            $consultation->setLocaux($this);
        }

        return $this;
    }

    public function removeConsultation(Consultation $consultation): self
    {
        if ($this->consultations->removeElement($consultation)) {
            // set the owning side to null (unless already changed)
            if ($consultation->getLocaux() === $this) {
                $consultation->setLocaux(null);
            }
        }

        return $this;
    }
    public function __toString()
    {
        return $this->localisation;
    }
}

File: src/Entity/Local.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package Entity;

import java.time.LocalDate;

/**
 *
 * @author donia
 */
public class Local {
      private int id;
    private String nom_block , nom_patient , nom_medecin ,localisation , image;

    public Local(int id, String nom_block, String nom_patient, String nom_medecin, String localisation , String image) {
        this.id = id;
        this.nom_block = nom_block;
        this.nom_patient = nom_patient;
        this.nom_medecin = nom_medecin;
        this.localisation = localisation;
        this.image = image ;
    }

    public Local() {
    }

    public Local(String nom_block, String nom_patient, String nom_medecin, String localisation , String image) {
        this.nom_block = nom_block;
        this.nom_patient = nom_patient;
        this.nom_medecin = nom_medecin;
        this.localisation = localisation;
        this.image = image ;
    }

    public String getImage() {
        return image;
    }

    public void setImage(String image) {
        this.image = image;
    }


    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getNom_block() {
        return nom_block;
    }

    public void setNom_block(String nom_block) {
        this.nom_block = nom_block;
    }

    public String getNom_patient() {
        return nom_patient;
    }

    public void setNom_patient(String nom_patient) {
        this.nom_patient = nom_patient;
    }

    public String getNom_medecin() {
        return nom_medecin;
    }

    public void setNom_medecin(String nom_medecin) {
        this.nom_medecin = nom_medecin;
    }

    public String getLocalisation() {
        return localisation;
    }

    public void setLocalisation(String localisation) {
        this.localisation = localisation;
    }

    @Override
    public String toString() {
        return "Local{" + "id=" + id + ", nom_block=" + nom_block + ", nom_patient=" + nom_patient + ", nom_medecin=" + nom_medecin + ", localisation=" + localisation + '}';
    }


     
}
